### PR TITLE
Submission HabitLogger

### PR DESCRIPTION
**Pull request by antonderegt**
Hi Reviewer,

Thanks for taking the time to look at my code!

PS. Codacy keeps complaining about the README.md, but it looks fine to me. I heard Codacy can have trouble with READMEs...

### CODE REVIEW

File: HabbitLogger.antonderegt/HabbitLogger/Program.cs [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+    }
+
+    void InsertRecord()
+    {
+        Console.Clear();
+        using var connection = new SqliteConnection(connectionString);
+        connection.Open();
+        SqliteCommand command = connection.CreateCommand();
+
+        Console.Write("New habit: ");
```
Comment (Doc415) [inline HabbitLogger.antonderegt/HabbitLogger/Program.cs:148]: Hey there i think you forgat to ask the quantity for the record to user. 
It is hard coded as 0 for all records.

File: HabbitLogger.antonderegt/HabbitLogger/Program.cs [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+namespace HabbitLogger;
+
+class Program
+{
+    private readonly string connectionString = @"Data Source=habitLogger.db";
+
+    static void Main(string[] args)
+    {
+        bool keepRunning = true;
+        Program program = new();
```
Comment (Doc415) [inline HabbitLogger.antonderegt/HabbitLogger/Program.cs:12]: Did you initialized the class inside itself on purpose?
You can directly call methods and fields if you make them static.

```
class Program
{
    static private readonly string connectionString = @"Data Source=habitLogger.db";

    static void Main(string[] args)
    {
        bool keepRunning = true;
        
       EnsureTableExists();
       SeedData();

        while (keepRunning)
        {
            ShowMenu();
            keepRunning = ProcessInput();
        }
    }

    static void EnsureTableExists()
    {
        using ......
```

**Review by Doc415 — CHANGES_REQUESTED** [review]

**Comment by Doc415** [comment]
> Hi Reviewer,
> 
> Thanks for taking the time to look at my code!
> 
> PS. Codacy keeps complaining about the README.md, but it looks fine to me. I heard Codacy can have trouble with READMEs...

Codacy is like a wife. It always find something to complain. You can just smile for the Readme.md ones.

File: HabbitLogger.antonderegt/HabbitLogger/Program.cs (hunk 2 above)
Comment (antonderegt) [reply HabbitLogger.antonderegt/HabbitLogger/Program.cs]: Thanks, now you mention it, it does make more sense to keep everything static and pass the connectionString around. Also makes the methods more flexible.

**Comment by antonderegt** [comment]
@Doc415 thanks for the review!

**Review by Doc415 — APPROVED** [review]
Thanks for your collaboration. Waiting for your next project.

**Comment by Doc415** [comment]
Thanks for your collaboration. Waiting for your next project.

File: HabbitLogger.antonderegt/HabbitLogger/Program.cs (hunk 2 above)
Comment (TheCSharpAcademy) [reply HabbitLogger.antonderegt/HabbitLogger/Program.cs]: @antonderegt Although it's ok for these beginner projects I'll point out the overusing static isn't good practice as it defeats the purpose of object oriented programming. Essentially it should only be used with classes that don't contain state. Not to mention concurrency issues. 😊

**Pull request closed without merging**